### PR TITLE
Migrating OLM testing pipeline from concourse to github-action

### PR DESCRIPTION
**Pull request by DanielePalaia**
This PR is part of the task to migrate our Concourse pipelines to github actions:

It is migrating the OLM pipeline in order to test our OLM bundle in the cluster-operator:
https://github.com/pivotal/rabbitmq-for-kubernetes-ci/blob/main/pipelines/olm-bundles/pipeline.yml

it can be useful to have a look to this guideline in order to see how to create and publish an OLM bundle:
https://devopstales.github.io/home/oml/

The test images are stored in the environment variable specified in:
  DOCKER_REGISTRY_SERVER: quay.io/rabbitmqoperator
  OLM_IMAGE: rabbitmq-for-kubernetes-olm-cluster-operator
  OLM_INDEX_IMAGE: rabbitmq-for-kubernetes-olm-cluster-operator-index

The action is always testing the latest cluster operator manifest (the last tagged version).
In case we run the action manually (through a workflow_dispatch) a generic 0.0.0 tag version is generated.

The cluster operator is deployed and configured in order to watch just  on "rabbitmq-system-olm" namespace so that the K8s cluster can be shared with the action which is testing the deployment using the carvel Package Repository (which is deploying the cluster operator and watching in the default "rabbitmq-system" namespace)

The lock is needed in order to share the cluster with the future messaging topology operator action.

The create-olm-package job is a bit confusing and overcomplicated but this will be improved when we will improve the conversion bundle script in python.

The new action and the old publishing action (openshift.yml) have been merged together.

Action result here: https://github.com/rabbitmq/cluster-operator/actions/runs/8052246499

Manifests are uploaded here:
https://github.com/rabbitmq/community-operators (operator-hub)
https://github.com/rabbitmq/community-operators-prod (openshift-marketplace)

Then a manual PR to openshfit repos is needed

### CODE REVIEW

File: .github/workflows/testing_and_publishing_OLM_bundle.yml [hunk 1]
```diff
@@ -0,0 +1,186 @@
+# action to test our operator lifecycle manager bundle
+# See https://github.com/rabbitmq/OLM-Package-Repo for more info.
+name: test-and-publish-OLM-bundle
+
+on:
+    push:
+      # Pattern matched against refs/tags
+      branches:
+        - "testing_olm"
```
Comment (PujaVad) [inline .github/workflows/testing_and_publishing_OLM_bundle.yml:9]: main?

Comment (DanielePalaia) [reply .github/workflows/testing_and_publishing_OLM_bundle.yml]: I will run this one on every release (I left the event on_push on my branch in the meantime in order to better test it)

**Comment by DanielePalaia** [comment]
The pipeline seems stable. I will merge it and test with the next tag release.

**Comment by DanielePalaia** [comment]
Taking in account Aitor review here: https://github.com/rabbitmq/OLM-Package-Repo/pull/8

**Comment by DanielePalaia** [comment]
I replaced git clone with checkout actions here too like suggested here: https://github.com/rabbitmq/messaging-topology-operator/pull/767. I will merge this one now to test it on the next operator release. We can come back with improments here when we improve the OLM Packaging script

**Pull request merged**